tox.ini

```diff
@@ -0,0 +1,68 @@
+[tox]
+envlist = py{35,36,37}-{pytest,setup}{,-all},docs,dev,flake8
+skipsdist = True
+requires = tox-venv
+           setuptools >= 30.0.0
+
+[testenv]
+usedevelop = True
+whitelist_externals = which
+passenv = HOME LC_ALL LC_CTYPE
+description =
+    setup: run tests with python setup.py
+    pytest: run tests with pytest
+deps =
+    pytest-cov
+    !all: pytest-astropy
+    !all: pytest-xdist
+    all: -rpip-requirements-dev
+conda_deps=
+    Cython
+    jinja2
+    numpy
+commands =
+    python --version
+    which python
+    pip freeze -l
+    pytest: pytest {posargs}
+    setup: python setup.py test {posargs}
+
+[testenv:dev]
+description = run with numpy and matplotlib dev versions
+deps =
+    -rpip-requirements-dev
+commands =
+    pip install git+https://github.com/numpy/numpy.git#egg=numpy
+    pip install git+https://github.com/matplotlib/matplotlib.git#egg=matplotlib
```

Comment:
Would it make more sense to move these to the `deps` section?

Reply:
This would conflict with pip-requirements-dev I think.

Reply:
Okay. I understand the reason for using a `pip-requirements` file but I wonder if you could have finer-grained control over dependencies by just including them specifically in the `tox.ini` file. Then you could prevent the conflict by using factors.